### PR TITLE
NO-JIRA: [ocp-]rhel-9.6: consume from RHEL 9.6 repos

### PR DESCRIPTION
**Pull request by jlebon**
RHEL 9.6 nightly content is already available to use so let's start tracking that instead for the 9.6 variants.

Note that OCP is approved to use that content.

While we're here, enact https://issues.redhat.com/browse/RHELPLAN-170883 where there was agreement that all container-related RPMs starting from 4.19+ will come from RHEL. This is prep for
https://github.com/openshift/enhancements/pull/1637.

### CODE REVIEW

File: manifest-ocp-rhel-9.6.yaml
Comment (travier) [inline manifest-ocp-rhel-9.6.yaml:1]:
```suggestion
# Manifest for OCP node based on RHEL 9.6
```

File: manifest-rhel-9.6.yaml
Comment (travier) [inline manifest-rhel-9.6.yaml:1]:
```suggestion
# Manifest for RHCOS based on RHEL 9.6
```

**Review by jbtrystram — APPROVED** [review]
/lgtm
/approve

**Comment by jlebon** [comment]
We're still waiting on final confirmation on this.

/hold

**Comment by jlebon** [comment]
We got confirmation on this now. For posterity, look at internal comments in https://url.corp.redhat.com/31bad51 and links there.

/hold cancel

Note that there is no 9.6 content yet on the mirrors. So for now we'll only be able to test this in earnest in the pipeline. I'll request mirrors to be added but since the content isn't actually on the CDN I'm not sure if it's possible.

**Comment by jlebon** [comment]
This one just needs a `/lgtm` now I think.

**Comment by mike-nguyen** [comment]
/lgtm

I'll have a follow-on version bump PR that I'll need to rebase on top of this. We also need to get the RHEL 9.6 content mirrored in prow so we can update our CI.

**Comment by jlebon** [comment]
> We also need to get the RHEL 9.6 content mirrored in prow so we can update our CI.

FTR, that request is tracked here: https://issues.redhat.com/browse/DPTP-4257

**Pull request merged**
